### PR TITLE
Support management of snmptrapd

### PR DESCRIPTION
**Pull request by elmobp**
#### Pull Request (PR) description
Add support for managing snmptrapd package/config

At the moment there is currently no option to stop shipping the snmpdtrapd package.

### CODE REVIEW

**Comment by ghoneycutt** [comment]
@elmobp Thanks for the contribution! Could you please add a test that shows what happens when the new parameter is set to `false`. It should show that the conditional resources do not exist as well as the related lines in the template.

**Comment by elmobp** [comment]
hey @ghoneycutt done! I had to do in multiple places given how the behaviour works. I also swapped the unless to a if statement to make the multiple condition easier to understand. Let me know if this is ok?

File: manifests/init.pp [hunk 1]
```diff
@@ -498,7 +502,7 @@
     }
   }
 
-  unless $facts['os']['family'] == 'Debian' {
+  if ($facts['os']['family'] != 'Debian') and ($manage_snmptrapd) {
```
Comment (ghoneycutt) [inline manifests/init.pp:505]: +1 

`not equals` is superior to `unless`

**Review by ghoneycutt — APPROVED** [review]
This is ready to merge, provided tests pass.

**Comment by ghoneycutt** [comment]
@elmobp very well done

**Comment by elmobp** [comment]
Test + squash

**Review by kenyon — APPROVED** [review]

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**